middleware/authenticateJWT.js

```diff
@@ -0,0 +1,19 @@
+import { RegisterRepository } from "../repositories/registerRepository.js";
+
+export function authenticateJWT(req, res, next) {
+  const authHeader = req.headers['auth'];
+  const token = authHeader && authHeader.split(' ')[1];
+
+  if (!token) {
+    return res.status(401).json('please provide token');
+  }
+
+  const registerRepository = new RegisterRepository();
```

Comment:
I would recommend moving token creation and verification logic into the auth service